addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JsoScanner.java

```diff
@@ -0,0 +1,137 @@
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.net.*;
+import java.nio.charset.*;
+import java.util.*;
+import net.htmlparser.jericho.*;
+import org.parosproxy.paros.*;
+import org.parosproxy.paros.core.scanner.*;
+import org.parosproxy.paros.network.*;
+import org.zaproxy.zap.extension.pscan.*;
+
+/** Java Serialized Objects (JSO) scanner. Detect the magic sequence and generate an alarm */
+public class JsoScanner extends PluginPassiveScanner {
+
+    /** Prefix for internationalized messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanalpha.jso.";
+
+    private static final byte[] JSO_BYTE_MAGIC_SEQUENCE = {(byte) 0xac, (byte) 0xed, 0x00, 0x05};
+    private static final String JSO_BASE_64_MAGIC_SEQUENCE = "rO0AB";
+    private PassiveScanThread parent;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // do nothing
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        checkJsoInHeaders(msg);
+
+        checkJsoInCookies(msg);
+
+        checkJsoInBody(msg);
+    }
+
+    private void checkJsoInBody(HttpMessage msg) {
+        byte[] startOfBody =
+                Arrays.copyOfRange(
+                        msg.getResponseBody().getBytes(), 0, JSO_BYTE_MAGIC_SEQUENCE.length);
+        String startOfBodyAsString =
+                new String(
+                        Arrays.copyOfRange(
+                                msg.getResponseBody().getBytes(),
+                                0,
+                                JSO_BASE_64_MAGIC_SEQUENCE.length()),
+                        StandardCharsets.UTF_8);
+        if (Arrays.equals(JSO_BYTE_MAGIC_SEQUENCE, startOfBody)
+                || hasJsoBase64MagicSequence(startOfBodyAsString)) {
+            Alert alert = buidAlert(msg, "");
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private void checkJsoInCookies(HttpMessage msg) {
+        for (HttpCookie cookie : msg.getResponseHeader().getHttpCookies(null)) {
+            String value = cookie.getValue();
+            if (!hasJsoBase64MagicSequence(value)) {
+                continue;
+            }
+
+            Alert alert = buidAlert(msg, cookie.toString());
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private boolean hasJsoBase64MagicSequence(String value) {
+        return value.startsWith(JSO_BASE_64_MAGIC_SEQUENCE);
+    }
+
+    private void checkJsoInHeaders(HttpMessage msg) {
+        for (HttpHeaderField header : msg.getResponseHeader().getHeaders()) {
+            String value = header.getValue();
+            if (!hasJsoBase64MagicSequence(value)) {
+                continue;
+            }
+
+            Alert alert = buidAlert(msg, header.toString());
+            parent.raiseAlert(getPluginId(), alert);
+        }
+    }
+
+    private Alert buidAlert(HttpMessage msg, String evidence) {
```

Comment:
This can be changed to `raiseAlert`, the returned alert is not used for anything other than raise the alert.